include/openthread/mdns.h

```diff
@@ -133,6 +133,24 @@ typedef otPlatDnssdService otMdnsService;
  */
 typedef otPlatDnssdKey otMdnsKey;
 
+/**
+ * Represents an mDNS entry iterator.
+ *
+ */
+typedef struct otMdnsIterator otMdnsIterator;
+
+/**
+ * Represents a host/service/key entry state.
+ *
+ */
+typedef enum otMdnsEntryState
+{
+    OT_MDNS_ENTRY_STATE_PROBING,    ///< Probing to claim the name.
+    OT_MDNS_ENTRY_STATE_REGISTERED, ///< Entry is successfully registered.
+    OT_MDNS_ENTRY_STATE_CONFLICT,   ///< Name conflict was detected.
+    OT_MDNS_ENTRY_STATE_REMOVING,   ///< Entry is being removed (sending "goodbye" announcements).
+} otMdnsEntryState;
+
```

Comment:
Why not include the state in otMdnsHost and otMdnsService?

Reply:
For this, we would need to include `otMdnsEntryState ` enum definition in `platform/dnss.h` and add `mState` in `otPlatDnssHost/Service` structures. These would not be used when registering host/service and by any of `otPlatDnss` APIs and can be confusing for platform-developers implementing the `otPlatDnss` APIs.

It made more sense to keep the state definition separate and in `mdns.h` and tied to `otMdns` APIs used for iterating over entries.